Adds aarch64_none_linux_gnu support

makestuffwork opened the pull request:
This is a straightforward adding aarch64_none_linux_gnu following the existing scheme.
There seems to be some unnecessary formatting rearrangements. Those are automatically done by the bazel style formatter embedded in my editor. I have good confidence in what it's doing.

hexdae commented on toolchain/toolchain.bzl:
```diff
@@ -35,6 +35,10 @@ target_constraints = {
         "armv7": ["@platforms//os:linux", "@platforms//cpu:armv7"],
     },
+    "aarch64-none-linux-gnu": {
+        "arm64": ["@platforms//os:linux", "@platforms//cpu:arm64"],
+        "aarch64": ["@platforms//os:linux", "@platforms//cpu:aarch64"],
```
nit: I think that [`arm64` is just an alias of `aarch64`](https://github.com/bazelbuild/platforms/blob/2af915c086d82ab0b233c8ff420ca72e128b8790/cpu/BUILD#L97-L100) so we can probably just use one here.

```starlark
alias(
    name = "arm64",
    actual = ":aarch64",
)
```

This is not blocking

hexdae commented:
Sorry, I added `aarch64-none-elf` recently to complement this and caused some conflicts, let me know if you want me to fix them or if you can rebase

makestuffwork commented:
> Sorry, I added `aarch64-none-elf` recently to complement this and caused some conflicts, let me know if you want me to fix them or if you can rebase

I rebased from your HEAD.

> This is great, do you mind adding a test in `test/aarch64-none-linux-gnu` if it's too much work, we can merge and I will add it later

Sorry, I'm really not skillful at adding this test. I did test it with my own project at https://github.com/google/pcie_lmt/blob/06ff42cb8724f75cd5fd6e081b03b3df3e42d2a3/BUILD#L60

hexdae commented:
This should be ready to go, thanks a lot for adding it. I am going to release soon
